editor/src/clj/editor/outline_view.clj

```diff
@@ -40,159 +49,234 @@
-                   root)
-        new-cache (assoc (map-filter (fn [[resource-node _]]
-                                       (contains? resource-node-set resource-node))
-                                     root-cache)
-                    active-resource-node [new-root active-outline hidden-node-outline-key-paths])]
-    (ui/user-data! raw-tree-view ::root-cache new-cache)
-    new-root))
-
-(defn- update-tree-view-selection!
-  [^TreeView tree-view selection old-selected-ids]
-  (binding [*programmatic-selection* true]
-    (sync-selection tree-view selection old-selected-ids)
-    tree-view))
-
-(defn- update-tree-view-root!
-  [^TreeView tree-view ^TreeItem root selection old-selected-ids]
-  (binding [*programmatic-selection* true]
-    (when (not (identical? (.getRoot tree-view) root))
-      (when root
-        (.setExpanded root true)
-        (.setRoot tree-view root)))
-    (sync-selection tree-view selection old-selected-ids)
-    tree-view))
-
-(g/defnk update-tree-view [^TreeView raw-tree-view ^TreeItem root selection]
-  (let [tree-view-ids (into #{}
-                            (map item->node-id)
-                            (.getSelectedItems (.getSelectionModel raw-tree-view)))]
-    (if (identical? (.getRoot raw-tree-view) root)
-      (if (= (set selection) (set (map :node-id (ui/selection raw-tree-view))))
-        raw-tree-view
-        (update-tree-view-selection! raw-tree-view selection tree-view-ids))
-      (update-tree-view-root! raw-tree-view root selection tree-view-ids))))
+(defn- auto-expand [selected-node-id->node-id-paths]
+  (->> selected-node-id->node-id-paths
+       (e/mapcat val)
+       (e/mapcat #(e/take-while coll/not-empty (iterate pop (pop %))))
+       (reduce
+         (fn [acc node-id-path]
+           (assoc! acc node-id-path true))
+         (transient {}))
+       persistent!))
+
+(defn decorate-outline [outline hidden-node-outline-key-paths outline-name-paths localization-state selection]
+  (letfn [(decorate [{:keys [node-id children] :as outline} node-id-path node-outline-key-path parent-reference?]
+            (let [node-id-path (conj node-id-path node-id)
+                  node-outline-key-path (if (empty? node-outline-key-path)
+                                          [node-id]
+                                          (if-some [node-outline-key (:node-outline-key outline)]
+                                            (conj node-outline-key-path node-outline-key)
+                                            node-outline-key-path))
+                  data (->> children
+                            (localization/sort-if-annotated localization-state)
+                            (mapv #(decorate %
+                                             node-id-path
+                                             node-outline-key-path
+                                             (or parent-reference? (:outline-reference? outline)))))
```

Comment:
Shouldn't this be `(:outline-reference? %)`?

Reply:
Doesn't look like it should, see the previous line of code that recursively called `decorate`:
https://github.com/defold/defold/pull/11343/files#diff-4c381bb725e8f7d75b6700638fe4d057dbaa9372f3d8cdf7f5bb6c94f104b518L137